### PR TITLE
Added ArmstrongNumber.java

### PR DESCRIPTION
**Pull request by rohangawhade**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Added a java code to check whether the input number is an Armstrong number or not.

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1217

### CODE REVIEW

**Comment by rohangawhade** [comment]
@siddharth25pandey  Kindly review my PR

File: Java/math/ArmstrongNumber.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	    int num = sc.nextInt();
+	    int res = num;
+	    int val=0;
+	    int len = length_number(num);
+	    while(res>0){
+	        int temp = res%10;
+	        val += Math.pow(temp, len);
+	        res = res/10;
+	    }
+	    if(num == val) System.out.println(num+" is armstrong");
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]:
```suggestion
	    if(num == val) System.out.println(num+" is an armstrong number.");
```

File: Java/math/ArmstrongNumber.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	    else System.out.println(num+" is not armstrong");
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]:
```suggestion
	    else System.out.println(num+" is not an armstrong number.");
```

File: Java/math/ArmstrongNumber.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	}
+}
+
+/*
+Sample input and output
+Example 1:
+Enter a number : 1002
+Number is not armstrong
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]: Edit sample I/O according to your code.

File: Java/math/ArmstrongNumber.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	    Scanner sc = new Scanner(System.in);
+	    System.out.print("Enter a number : ");
+	    int num = sc.nextInt();
+	    int res = num;
+	    int val=0;
+	    int len = length_number(num);
+	    while(res>0){
+	        int temp = res%10;
+	        val += Math.pow(temp, len);
+	        res = res/10;
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]:
```suggestion
	        res = res / 10;
```

File: Java/math/ArmstrongNumber.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    }
+	public static void main(String[] args) {
+	    Scanner sc = new Scanner(System.in);
+	    System.out.print("Enter a number : ");
+	    int num = sc.nextInt();
+	    int res = num;
+	    int val=0;
+	    int len = length_number(num);
+	    while(res>0){
+	        int temp = res%10;
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]:
```suggestion
	        int temp = res % 10;
```

File: Java/math/ArmstrongNumber.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+{
+    static int length_number(int num){
+        return String.valueOf(num).length();
+    }
+	public static void main(String[] args) {
+	    Scanner sc = new Scanner(System.in);
+	    System.out.print("Enter a number : ");
+	    int num = sc.nextInt();
+	    int res = num;
+	    int val=0;
```
Comment (atarax665) [inline Java/math/ArmstrongNumber.java]:
```suggestion
	    int val = 0;
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: Java/math/ArmstrongNumber.java (hunk 6 above)
Comment (rohangawhade) [reply Java/math/ArmstrongNumber.java]: @atarax665  Made the suggested changes.

**Comment by HarshCasper** [comment]
LGTM!

**Pull request merged**